### ISSUE TITLE
[BUG] Windows pip wheels only up to version 0.9.0

### ISSUE THREAD

**Issue by 2-5** [alan-turing-institute/sktime#2715]
The documentation seems to suggest that Windows is supported, but the latest Windows `pip` binaries (from https://pypi.org/simple/sktime/) are from version 0.9.0, and only up to Python 3.8.

Is `sktime` only supported under `conda` and not when using plain Python with `pip`?

I see under another issue that `prophet` is causing problems with Python 3.10.  Maybe there could be a minimal `sktime` version which works under all Python versions and all OSes, and put `prophet` and other difficult dependencies into an extra version.

**Comment by fkiraly** [comment]
> I see under another issue that prophet is causing problems with Python 3.10. Maybe there could be a minimal sktime version which works under all Python versions and all OSes, and put prophet and other difficult dependencies into an extra version.

The 3.10/`prophet` issue and possible solutions are discussed here:
https://github.com/alan-turing-institute/sktime/issues/2079

There is a potential simple solution where we have individual estimators requiring a version but still shipping with `sktime`, which some other core devs are against (@lmmentel can give his opinion 😄 )

The alternative (internal) options that ensure 3.10 compatibility would be: splitting `sktime`, as you say, or removing `prophet` altogether.

**Comment by fkiraly** [comment]
> The documentation seems to suggest that Windows is supported, but the latest Windows `pip` binaries (from https://pypi.org/simple/sktime/) are from version 0.9.0, and only up to Python 3.8.

Since 0.10.0, the wheels are no longer operating system specific, because the need to compile has vanished with the C dependencies. They should work for windows, since they work for "any" OS.

**Comment by fkiraly** [comment]
(@lmmentel, as other release manager, could you kindly confirm whether what I said about windows is correct?)

**Comment by 2-5** [comment]
Sorry, my mistake. I am on Windows/Python 3.10 and when I did a `pip install sktime` it downloaded version `0.9.0` and tried to install it, but failed with a weird error. Since that was the last one with binary wheels, I assumed it was somehow related.

But great news! Based on your comment about not needing binaries I did a `pip install sktime --ignore-requires-python` and it worked and installed `sktime-0.11.4` despite me being on Python 3.10.

I tried the first example from the get started page and it works correctly.

The second one also works, but you need to update the code on your page, the current version fails with an import error, since it looks like `TimeSeriesForestClassifier` was moved from `sktime.classification.kernel_based` to `sktime.classification.interval_based`

On a more general note, why forcefully limit future version support? I see `sktime` limits `numpy < 1.22`, `Python < 3.10` among other things (for example `pandas < 1.5.0`, but that version doesn't even exist yet). I have `numpy 1.23.4+mkl` and the sample worked.

I think it would be better to not limit upwards unless it's known for a fact that it's very incompatible, and not just limit in a precautionary way.

**Comment by fkiraly** [comment]
> The second one also works, but you need to update the code on your page, the current version fails with an import error, since it looks like `TimeSeriesForestClassifier` was moved from `sktime.classification.kernel_based` to `sktime.classification.interval_based`

@2-5, which page are you talking about? On the readme, it seems to correctly say `interval_based`?

**Comment by fkiraly** [comment]
> On a more general note, why forcefully limit future version support?

If not limiting it, automatic version resolution might automatically install python 3.10?
Which is not fully supported, and hence may produce errors in a "default install", to avoid this, 3.9 or lower would now have to be expressly requested.

**Comment by 2-5** [comment]
> @2-5, which page are you talking about? On the readme, it seems to correctly say `interval_based`?

https://www.sktime.org/en/latest/get_started.html#time-series-classification

**Comment by 2-5** [comment]
> If not limiting it, automatic version resolution might automatically install python 3.10? 

I guess you mean when you do `conda install sktime` it might install Python 3.10? I never used `conda`, I wasn't aware it can install Python too, so I guess you are right.

`pip` however will never install or upgrade Python, that's not something it can do.

I did a little googling on best practices, and I found this (long) blog post on this issue from Henry Schreiner, Python Packaging Authority member:

https://iscinumpy.dev/post/bound-version-constraints/

> Bound version constraints (upper caps) are starting to show up in the Python ecosystem. This is causing real world problems with libraries following this recommendation, and is likely to continue to get worse; this practice does not scale to large numbers of libraries or large numbers of users. In this discussion I would like to explain why always providing an upper limit causes far more harm than good even for true SemVer libraries, why libraries that pin upper limits require more frequent updates rather than less, and why it is not scalable. After reading this, hopefully you will always consider every cap you add, you will know the (few) places where pinning an upper limit is reasonable, and will possibly even avoid using libraries that pin upper limits needlessly until the author updates them to remove these pins.

> ...

> **Never provide an upper cap to your Python version. I generally will not use a library that has an upper cap to the Python version**

> ...

> TL;DR
> Capping dependencies has long term negative effects, especially for libraries, and should never be taken lightly. A library is not installed in isolation; it has to live with other libraries in a shared environment. Only add a cap if a dependency is known to be incompatible or there is a high (>75%) chance of it being incompatible in its next release. Do not cap by default - capping dependencies makes your software incompatible with other libraries that also have strict lower limits on dependencies, and limits future fixes. Anyone can fix a missing cap, but users cannot fix an over restrictive cap causing solver errors. It also encourages hiding issues until they become harder to fix, it does not scale to larger systems, it limits your ability to access security and bugfix updates, and some tools (Poetry) force these bad decisions on your downstream users if you make them. **Never cap Python, it is fundamentally broken at the moment.** Also, even packing capping has negative consequences that can produce unexpected solves.
> ...
> If you need a TL;DR for the TL;DR, I’ll just [quote](https://snarky.ca/why-i-dont-like-semver/) Python Steering Council Member and packaging expert Brett Cannon:
>> Libraries/packages should be setting a floor, and if necessary excluding known buggy versions, but otherwise don’t cap the maximum version as you can’t predict future compatibility

**Comment by 2-5** [comment]
@fkiraly Sorry, I forgot to tag you so that you can see my replies since this is a closed issue.

**Comment by fkiraly** [comment]
@2-5, yes, indeed I did not see it due to that.

I would suggest, let's continue discussing here:
https://github.com/alan-turing-institute/sktime/discussions/2651
which seems to be the "canonical" place.

I transplanted our discussion on the matter of python version and version bounds.
